str.go

```diff
@@ -70,3 +73,45 @@ func (str *SignedTreeRoot) serialize() []byte {
 
 	return strBytes
 }
+
+func (str *SignedTreeRoot) MarshalJSON() ([]byte, error) {
```

Comment:
I think we should move all of the JSON marshaling functions to the server. I think we should keep this library only for the data structure.